cmd/query/app/apiv3/http_gateway_test.go

```diff
@@ -37,7 +38,11 @@ func setupHTTPGatewayNoServer(
 
 	q := querysvc.NewQueryService(v1adapter.NewTraceReader(gw.reader),
 		&dependencyStoreMocks.Reader{},
-		querysvc.QueryServiceOptions{},
+		querysvc.QueryServiceOptions{
+			Adjuster: adjuster.Func(func(*model.Trace) {
```

Comment:
@yurishkuro if the adjuster isn't set then there's a nil pointer dereference because we assume it to be set. As an alternative, we can perform a nil check before trying to use the adjuster.

Reply:
NewQueryService already has the nil check
```go
	if qsvc.options.Adjuster == nil {
		qsvc.options.Adjuster = adjuster.Sequence(StandardAdjusters(defaultMaxClockSkewAdjust)...)
	}
```

Reply:
@yurishkuro sorry yes you're absolutely right. I digged into this a bit more and it looks like its because the adjusters assume a non-nil `span.Process.Tags`. For example - https://github.com/jaegertracing/jaeger/blob/main/model/adjuster/clockskew.go#L70. If `span.Process` is nil (which it is in these tests), then there's a nil pointer dereference trying to get `span.Process.Tags`. Is that a valid assumption? Or should we open a separate PR to handle the case where `span.Process` is nil?

Reply:
> `model.KeyValues(span.Process.Tags).FindByKey("ip")`

This assumed that Process is not nil, it will still work for Tags being nil.

It is technically invalid to have a span with nil Process at this stage. Even if the data arrives like that externally, the otlp2model should still create an empty Process and the sanitizers will certainly set Process.ServiceName to a non-empty string.

In summary, I don't think we should be putting nil checks all over the place.

Reply:
sounds good - i'll change these tests to pass in a parent process then